contracts/brush/proc_macros/lib.rs

```diff
@@ -0,0 +1,259 @@
+        #attrs
+        #[ink_lang::contract]
+        #module
+    };
+    result.into()
+}
+
+#[proc_macro_attribute]
+pub fn trait_definition(_attrs: TokenStream, _input: TokenStream) -> TokenStream {
+    let input: TokenStream2 = _input.clone().into();
+    let attrs: TokenStream2 = _attrs.into();
+    let trait_item = parse_macro_input!(_input as ItemTrait);
+
+    let locked_file = get_locked_file();
+    let mut hash_map = load_hash_map(&locked_file);
+    put_trait(&mut hash_map, trait_item);
+    save_hash_map_and_unlock(locked_file, hash_map);
+
+    let code = quote! {
+        #attrs
+        #[ink_lang::trait_definition]
+        #input
+    };
+    code.into()
+}
+
+struct ImplTrait {
+    contract: syn::Ident,
+    traits: Vec<(syn::Ident, Option<syn::Ident>)>,
+}
+
+fn parse_impl_trait(stream: TokenStream2) -> ImplTrait {
+    let mut iter = stream.into_iter();
+    let mut impl_trait;
+    if let TokenTree::Ident(ident) = iter.next().expect("Empty token stream") {
+        impl_trait = ImplTrait{
+            contract: ident,
+            traits: vec![]
+        };
+    } else {
+        panic!("First token is not struct ident");
+    }
+    for item in iter {
+        if let TokenTree::Punct(_) = item {
+            continue
+        } else if let TokenTree::Ident(trait_ident) = item {
+            impl_trait.traits.push((trait_ident, None));
+        } else if let TokenTree::Group(trait_group) = item {
+            if let Some(pair) = impl_trait.traits.last_mut() {
+                let internal_trait = &mut pair.1;
+                if internal_trait.is_none() {
+                    *internal_trait = Some(syn::parse2::<syn::Ident>(trait_group.stream()).expect("Can't find ident of internal trait"));
+                } else {
+                    panic!("External trait already contains internal");
+                }
+            } else {
+                panic!("Internal trait before external");
+            }
+        }
+    }
+    impl_trait
+}
+
+#[proc_macro]
+pub fn impl_trait(_item: TokenStream) -> TokenStream {
+    let impl_trait = parse_impl_trait(TokenStream2::from(_item));
+
+    let contract_ident = impl_trait.contract;
+
+    let locked_file = get_locked_file();
+    let traits = load_hash_map(&locked_file);
+    locked_file.unlock().expect("Can't remove exclusive lock in impl_trait");
+    let mut impls: Vec<TokenStream2> = vec![];
+
+    // println!("{:?}", impl_trait.clone());
+    for (external_trait, internal) in impl_trait.traits.iter() {
+        let trait_methods: Vec<_> = traits.get(&external_trait.to_string())
+            .expect("Can't find definition of external trait").into_iter()
+            .map(|method| TokenStream2::from_str(method).expect("Can't parse definition of external trait"))
+            .map(|stream| syn::parse2::<syn::TraitItemMethod>(stream).expect("Can't parse external method"))
+            .collect();
+
+        let implementations = trait_methods.into_iter().map(|item| {
```

Comment:
Can we break implementation down into smaller functions? It would be more clear what each code block does and will help to troubleshoot bugs, if there are.

Reply:
Okey, we will do that during another PR where we will add comments with description of macro